dips/dip-158.md

```diff
@@ -0,0 +1,805 @@
+| currency | enum | Y | One of the supported on-chain currency types, e.g. XUS|
+| action | [Action enum](#action-enum) | Y | Populated in the request.  This value indicates the requested action to perform. For immediate capture and P2P transfer, "charge" is used.  For auth and capture, "auth" and "capture" should be used. "capture" can only be performed after a valid "auth" |
+| valid_until | uint | N | Unix timestamp indicating until when the payment can be authorized. Once this time has been reached, an attempt to authorize and lock the funds will be rejected. One should consider limiting this period to several minutes (e.g. 30 minutes) which should sufficve for the customer to approve or reject the payment request|
+| timestamp | uint | Y | Unix timestamp indicating the time that the payment Command was created.|
+
+```
+{
+    "amount": 100000000,
+    "currency": "XUS",
+    "action": "auth",
+    "valid_until": 74000,
+    "timestamp": 72322,
+}
+```
+### Action Enum
+The following values are allowed for the Action Enum
+* `charge`
+* `auth`
+* `capture`
+
+## PaymentStatusObject
+| Field | Type | Required? | Description |
+|-|-|-|-|
+| status | [Status enum](#status-enum) | Y | Status of the payment|
+| abort_code |[AbortCode enum](#abortcode-enum)| N | In the case of an `abort` status, this field may be used to describe the reason for the abort. Represents the error code of the corresponding error|
+| abort_message | str | N | Additional details about this error.  To be used only when `code` is populated |
+```
+"status": {
+  "status": "abort",
+  "abort_code": "void",
+  "abort_message": "abort message"
+}
+```
+
+### Status Enum
+* `none` - Initial state
+* `ready_for_settlement` - Payment is ready for settlement on chain. This means the actor has performed all required tasks (on its end). Note that the actual settlement is done by the wallet (sender), so for the receiver this status means it is willing to accept the payment.
+* `authorized` - Funds can be locked. This means the actor has performed all required tasks (on its end) and funds can be locked for a future capture. Note that the actual lock is done by the wallet (sender), so for the receiver this status means it is willing to accept the payment. Status is relevant for auth/capture payments only
+* `needs_init_data`- This means the actor requires basic data to initialize the payment. For the receiver, this status means that the payer data is required. For the sender, it means that the merchant data is required
+* `abort` - Payment is cancelled
+
+### Allowed status transitions
+| Current Status | Next Status | Transition Comments |
+|-|-|-|
+| none | needs_init_data |Initial transition. Most common for P2M scenarios in which the actor requires additional data to initialize the payment|
```

Comment:
I think there's actually two states:

- merchant: waiting_for_customer_vasp
- customer_vasp: needs_payment_data

trying to merge two state machines into one leaves a bit of ambiguity.

Reply:
Removed PaymentStatusObject from DIP